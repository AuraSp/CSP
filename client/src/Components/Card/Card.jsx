import React, { useState, useEffect, Fragment } from 'react';
import HistoryDiagram from '../../Diagram/HistoryDiagram';

// import '../../../Styles/Card/card.css';

function Card({ info, stocks, time }) {

    //MODAL FOR DIAGRAM
    const [popUp, setPopup] = useState(false);

    //ERROR STATE VARIABLE FOR STOCK(if no data exists or not enough)
    const [errMsg, setErrMsg] = useState('');

    //STATE VARIABLE FOR DIAGRAM
    const [diagramData, setDiagramData] = useState();

    //UPDATING STOCK DATA(according to date)
    useEffect(() => {
        if (stocks.s === "no_data") {
            setErrMsg('Using current dates! No data to show');
            const postToLogs = {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({
                    level: 'warn',
                    message: 'Object is empty for choosen data'
                })
            };
            fetch('http://localhost:5500/api/postLogs/', postToLogs);
        } else if (stocks.s === "ok" && time.length === 1) {
            setErrMsg('Not enough stock data to render! Choose at least 10 days gap');
            const postToLogs = {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({
                    level: 'warn',
                    message: 'Object has not enough stock info to construct diagram'
                })
            };
            fetch('http://localhost:5500/api/postLogs/', postToLogs);
        } else {
            let temp = Array.from({ length: time.length }, (_, i) => ({ Date: stocks.t[i], Open: stocks.o[i], High: stocks.h[i], Low: stocks.l[i], Close: stocks.c[i], Volume: stocks.v[i] }));
            setDiagramData(temp);
        }
    }, [time]);

    //INTERACTIVE BUTTON FOR DIAGRAM DISPLAY
    const getChart = () => {
        setPopup(true);
    };


    return (
        <div className='row mx-auto'>
            {Object.entries(info).length === 0 ?
                (
                    <p className='rounded text-center text-light fs-5 mx-auto p-4 my-3 loading'>Company with that name does not exist!</p>
                )
                :
                (
                    <Fragment>

                        <img src={info.logo ? info.logo : 'No logo...'} alt='Company Logo' className='col-lg-12 col-md-12 col-sm-12 img-thumnbnail p-0 w-25 rounded mx-auto my-3' />
                        <ul className='col-lg-12 col-md-12 col-sm-12 d-flex flex-row flex-wrap justify-content-center list-unstyled my-1 list' data-testid='cardlist'>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center'><button className='btn border-0 text-light' onClick={getChart}>{info.name}</button></li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1  border-danger d-flex justify-content-center align-items-center text-light'>{info.country}</li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center text-light'>{info.currency}</li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center'>{info.weburl ? <a href={info.weburl} className='text-decoration-none text-light'>{info.weburl}</a> : 'No website url'}</li>
                        </ul>
                        {popUp && (
                            !diagramData ? errMsg : <HistoryDiagram
                                diagramData={diagramData
                                } />
                        )}
                    </Fragment>
                )
            }
        </div >
    )

}

export default Card