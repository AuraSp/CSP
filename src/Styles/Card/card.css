@import url("https://fonts.googleapis.com/css2?family=Kalam&family=Patrick+Hand&display=swap");
.list li {
  font-family: Kalam;
  background: rgba(34, 34, 34, 0.8666666667);
  border-radius: 1em;
  height: 6em;
  box-shadow: 0 0 10px rgba(34, 34, 34, 0.8666666667);
}/*# sourceMappingURL=card.css.map */