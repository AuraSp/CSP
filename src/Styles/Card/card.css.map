{"version":3,"sources":["../Variables/variables.scss","card.scss","card.css"],"names":[],"mappings":"AACQ,8FAAA;ACEJ;EACI,kBAAA;EACA,0CAAA;EACA,kBAAA;EACA,WAAA;EACA,mDAAA;ACDR","file":"card.css"}