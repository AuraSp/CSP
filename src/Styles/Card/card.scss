@import '../Variables/variables.scss';

.list {
    li {
        font-family: map-get($font-families, Kalam);
        background: map-get($background-colors, card-background);
        border-radius: map-get($border-radius, "small");
        height: map-get($heights, card-tiles);
        box-shadow: map-get($shadows, cardList-shadow);
    }
}