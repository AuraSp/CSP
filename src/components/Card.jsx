import React from 'react'

function Card() {
    return (
        <div className="row my-4 mx-auto">
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
            <div className="col-lg-2 col-md-2 col-sm-4 border border-2 rounded text-center my-1">Title Test</div>
        </div>
    )
}

export default Card