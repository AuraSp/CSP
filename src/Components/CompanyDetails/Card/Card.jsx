import React, { useState, useEffect, Fragment } from 'react';
import HistoryDiagram from '../../Diagram/HistoryDiagram';
import logger from '../../../Utils/logger.js';

import '../../../Styles/Card/card.css';

function Card({ errMsg, inputValue, info, loading, getStock, changeDate }) {

    const [popupOpen, setPopupOpen] = useState(false);
    const [diagramData, setDiagramData] = useState();


    useEffect(() => {
    }, [diagramData])



    const getChart = () => {
        if (changeDate === undefined) {
            logger.error('No timestamp for construction. Maybe data is not fetched fully or choosed recent one?')
        } else {
            logger.info('Constructing new data for diagram')

            let temparr = Array.from({ length: changeDate.length }, (_, i) => ({ Date: changeDate[i], Open: getStock.o[i], High: getStock.h[i], Low: getStock.l[i], Close: getStock.c[i], Volume: getStock.v[i] }))

            setDiagramData(temparr)
            setPopupOpen(true)
        }
    }


    return (
        <div className='row mx-auto'>
            { info === undefined ?
                (
                    <p className='rounded text-center text-light fs-5 mx-auto p-4 my-3 loading'>Company with that name does not exist!</p>
                )
                :
                (
                    <>

                        <img src={info.logo ? info.logo : 'Loading logo...'} alt='Company Logo' className='col-lg-12 col-md-12 col-sm-12 img-thumnbnail p-0 w-25 rounded mx-auto my-3' />
                        <ul className='col-lg-12 col-md-12 col-sm-12 d-flex flex-row flex-wrap justify-content-center list-unstyled my-1 list' data-testid='cardlist'>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center'><button className='btn border-0 text-light' onClick={getChart}>{info.name}</button></li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1  border-danger d-flex justify-content-center align-items-center text-light'>{info.country}</li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center text-light'>{info.currency}</li>
                            <li className='col-lg-4 col-md-5 col-sm-12 border m-1 border-danger d-flex justify-content-center align-items-center'><a href={info.weburl} className='text-decoration-none text-light'>{info.weburl}</a></li>
                        </ul>
                        {changeDate === undefined ?
                            <p className='rounded text-center text-light fs-5 mx-auto p-4 my-3 loading'>{errMsg}</p>
                            :
                            popupOpen &&
                            <HistoryDiagram
                                loading={loading}
                                diagramData={diagramData}
                                inputValue={inputValue}
                            />
                        }
                    </>
                )
            }
        </div >
    )

}

export default Card