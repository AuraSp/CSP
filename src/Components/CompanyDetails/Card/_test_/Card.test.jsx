import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import Card from '../Card'

describe('testing component rendering', () => {

  test('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});