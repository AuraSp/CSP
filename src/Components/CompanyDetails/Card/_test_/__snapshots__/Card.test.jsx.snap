// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row mx-auto"
>
  <p
    className="rounded text-center text-light fs-5 mx-auto p-4 my-3 loading"
  >
    Company with that name does not exist!
  </p>
</div>
`;

exports[`testig component rendering renders correctly 1`] = `
<div
  className="row mx-auto"
>
  <p
    className="rounded text-center text-light fs-5 mx-auto p-4 my-3 loading"
  >
    Company with that name does not exist!
  </p>
</div>
`;

exports[`testing component rendering renders correctly 1`] = `
<div
  className="row mx-auto"
>
  <p
    className="rounded text-center text-light fs-5 mx-auto p-4 my-3 loading"
  >
    Company with that name does not exist!
  </p>
</div>
`;
